imports/api/questions.js

```diff
@@ -308,8 +303,12 @@ Meteor.methods({
 
     if (user.isStudent(question.courseId)) question.approved = false
 
+    // Question cannot be both public and private
+    if (question.public && question.private) {
```

Comment:
I would go the other way around and prioritize private = true; Check if private:true and set public false